src/screens/Dashboard.js

```diff
@@ -5,95 +5,195 @@
+ * @readonly
+ * @enum {string}
+ */
+const listOption = {
+  tokens: 'tokens',
+  nanoContracts: 'nanoContracts',
+};
+
+/**
+ * @param {listOption} currList the list component selected to be rendered.
+ * @returns {boolean} `true` if tokens list is selected, `false` otherwier.
+ */
+const isTokensSelected = (currList) => currList === listOption.tokens;
 
-class Dashboard extends React.Component {
-  static navigatorStyle = { tabBarVisible: false }
+/**
+ * @param {listOption} currList the list component selected to be rendered.
+ * @returns {boolean} `true` if nanoContracts list is selected, `false` otherwier.
+ */
+const isNanoContractsSelected = (currList) => currList === listOption.nanoContracts;
 
-  onItemPress = (item) => {
-    // Check if the token balance is already loaded
-    const tokenBalanceStatus = get(this.props.tokensBalance, `${item.uid}.status`, 'loading');
+export const Dashboard = () => {
+  const {
+    tokens,
+    tokensBalance,
+    selectedToken,
+    tokensMetadata,
+  } = useSelector(getTokensState);
+  const [currList, selectList] = useState(listOption.tokens);
+  const navigation = useNavigation();
+  const dispatch = useDispatch();
 
-    if (tokenBalanceStatus === 'loading') {
+  const onTokenPress = (token) => {
+    const status = getTokensBalanceStatus(tokensBalance, token);
+
+    if (isTokensBalanceLoading(status)) {
       return;
     }
 
-    if (tokenBalanceStatus === 'failed') {
+    if (isTokensBalanceFailed(status)) {
       // If the token balance status is failed, we should try again
-      this.props.getBalance(item.uid);
+      dispatch(tokenFetchBalanceRequested(token.uid))
       return;
     }
 
-    this.props.updateSelectedToken(item);
-    this.props.navigation.navigate('MainScreen');
+    dispatch(updateSelectedToken(token));
+    navigation.navigate('MainScreen');
   }
 
-  render() {
-    const ManualInfoButton = () => (
-      <SimpleButton
-        title={t`Register token`}
-        onPress={() => this.props.navigation.navigate('RegisterToken')}
-      />
-    );
-
-    const Header = () => (
-      <HathorHeader
-        title={t`TOKENS`}
-        rightElement={<ManualInfoButton />}
-      />
-    );
-
-    return (
-      <View style={{ flex: 1 }}>
-        <ShowPushNotificationTxDetails navigation={this.props.navigation} />
-        <AskForPushNotification navigation={this.props.navigation} />
-        <AskForPushNotificationRefresh />
+  return (
+    <Wrapper>
+      <ShowPushNotificationTxDetails navigation={navigation} />
+      <AskForPushNotification navigation={navigation} />
+      <AskForPushNotificationRefresh />
+      <DashBoardHeader>
+        <TwoOptionsToggle
+          options={{
+            first: { value: 'Tokens', onTap: () => selectList(listOption.tokens) },
+            second: { value: 'Nano Contracts', onTap: () => selectList(listOption.nanoContracts) }
+          }}
+          defaultOption='first'
+        />
```

Comment:
I'm missing here the feature toggle to hide the nano contracts tab and show the original and current screen.

Reply:
Sure, I've opened [this PR](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/483) to introduce the feature toggle. I believe the feature toggle was created after I prepare this PR in the past. I can join the feature toggle PR in this PR, what you prefer?

Reply:
We can't merge this PR without the feature toggle, so it's better to add in this PR.

Reply:
Ok.

Reply:
Done.